vjudge/status.py

```diff
@@ -0,0 +1,44 @@
+
+
+status_keywords = ['accept', 'answer', 'compilation', 'compile', 'error',
+    'exceed', 'limit', 'memory', 'runtime', 'time', 'wrong']
+
+# Fetch status every 3 seconds
+# Should run as daemon
+def status_daemon():
+    while True:
+        try:
+            raw_status = requests.get(status_url, timeout=5).text.encode("utf-8")
+            raw_status = json.loads(raw_status)['data']
+
+            refined_status = {}
+            for status in raw_status:
+                sid = status[0]
+                result = status[3]
+                refined_status[sid] = result
+
+            submissions = Submission.objects.filter(status__in=[Submission.WAIT, Submission.JUDGING])
+
+            for submission in [s for s in submissions if s.problem.judge_source == Problem.OTHER and s.other_judge_sid]:
```

Comment:
Maybe you can try [Changron's code](https://github.com/bruce3557/NTHUOJ_web/pull/88/files#diff-b1d7116ac5e617eac43cc27e0777d412R40)
[issue on stack overflow](http://stackoverflow.com/questions/1981524/django-filtering-on-foreign-key-properties)
This django filter can query attribute of foreign key.